### ISSUE TITLE
[cncli.sh] support syncing to 100% and then exit

### ISSUE THREAD

**Issue by TrevorBenson** [cardano-community/guild-operators#1752]
**Describe the feature**
Have cncli.sh support an option to sync to 100% and then exit. This is available as flag `--no-service` in cncli's sync subcommand.  This feature would allow a user to run a oneshot type systemd service unit for cncli sync.

**Expected behavior**
Without any additional flags provided the sync should continue to work as a service as cncli.sh does today.

When the cncli.sh gets requested to not be a service for the `sync` subcommand it will exit like the cncli parent tool would:
```
 2024-04-13T21:02:27.328Z INFO  cncli::nodeclient::sync > get_intersect_blocks took: 220.997µs
 2024-04-13T21:02:27.336Z INFO  cncli::nodeclient::sync > block 1985108 of 1985109:  99.99% sync'd
 2024-04-13T21:02:27.337Z INFO  cncli::nodeclient::sync > block 1985109 of 1985109: 100.00% sync'd
 2024-04-13T21:02:27.337Z INFO  cncli::nodeclient::sync > Exiting...
```

**Comment by Scitz0** [comment]
Hey, what would be the use case for this?
Ie you mentioned the ability to run a oneshot type service, but I struggle to see in what case this would be useful. 

Its not a problem to add, just interested in its use :)

**Comment by TrevorBenson** [comment]
> Hey, what would be the use case for this? Ie you mentioned the ability to run a oneshot type service, but I struggle to see in what case this would be useful.
> 
> Its not a problem to add, just interested in its use :)

Mostly a container-based use case, or at least my preference. Under Systemd I use service units for sync, leaderlog and validate and timer units on Leaderlog and Validate. The service units in my case run `podman exec -it <container> /opt/cardano/cnode/scripts/cncli.sh -u <command>`. I've just been lazily modifying the cncli.sh script to include ` ${CNCLI} sync --no-service` manually in **cncliSync** function, and then forgetting to open a ticket and submit code so it is an option instead of hardcoded.

--------
TLDR

The original reason was a year or more ago I noticed `cncli sync` seemed to keep the CPU core under load, which I suspected was a bug but admittedly did not open an issue. At that time I believe it was floating around 90% or greater, at least from what I recall. Since you asked I just checked on my BP for the preview testnet, it seems relatively low at 0-1%. I haven't checked this against my mainnet node however, which is where the original CPU load behavior was observed.

If this is no longer the case I could make sync a persistent service again, and we could just close this request unless others would also like to have a non daemon mode available.

**Comment by Scitz0** [comment]
Ok thanks for explaining reason. I think the cpu thing is/was a bug related to corrupt cncli db. Normally cpu load should be close to negligible.

**Comment by TrevorBenson** [comment]
I'll double check my prod BP on mainnet. Short of seeing unexpected loads I'll close this request later this evening.

**Comment by TrevorBenson** [comment]
I am definitely not seeing the high CPU usage on any network, including production mainnet. On mainnet a Ryzen 7 5700 syncing of blocks 4573536 -  10207326 was as low as 2% at times, never spiked above 11%, and averaged 7% for the full sync. Afterwards it settled down to below 3% between blocks.

If I see the prior behavior return I'll just replace the cncli db. The load is completely acceptable even in a k8s application as a container instead of a scheduled cronjob, so I'm going to close this request.

Thanks @Scitz0
